src/request_handler_static.cpp

```diff
@@ -21,45 +21,56 @@
 namespace http {
 namespace server {
 
-request_handler_static::request_handler_static(const std::string& static_file_root) {
-	static_file_root_ = static_file_root;
-}
+request_handler_static::request_handler_static(const server_options* server_options_) 
+  : server_options_(server_options_){}
```

Comment:
Good change! Much more robust way of handling the different options. We'll just have to change some of the unit tests